control/pid_longitudinal_controller/src/pid_longitudinal_controller.cpp

```diff
@@ -428,19 +444,27 @@ PidLongitudinalController::ControlData PidLongitudinalController::getControlData
   // current velocity and acceleration
   control_data.current_motion.vel = m_current_kinematic_state.twist.twist.linear.x;
   control_data.current_motion.acc = m_current_accel.accel.accel.linear.x;
+  control_data.interpolated_traj = m_trajectory;
 
   // nearest idx
-  const size_t nearest_idx = motion_utils::findFirstNearestIndexWithSoftConstraints(
-    m_trajectory.points, current_pose, m_ego_nearest_dist_threshold, m_ego_nearest_yaw_threshold);
-  const auto & nearest_point = m_trajectory.points.at(nearest_idx).pose;
+  const auto current_interpolated_pose =
+    calcInterpolatedTrajPointAndSegment(control_data.interpolated_traj, current_pose);
+  const size_t nearest_idx = current_interpolated_pose.second + 1;
```

Comment:
`nearest_idx` is not always equivalent to `segment_idx + 1`. 
But the `segment_idx + 1` will be `nearest_idx` when the point is inserted in the `interpolated_traj`.

I guess this variable is not used anywhere?

Reply:
Thank you for the review Horibe-san. Because it will be inserted into interpolated_traj, I set current_interpolated_pose.second + 1` to `nearest_idx` as below:

```
  // Insert the interpolated point
  control_data.interpolated_traj.points.insert(
    control_data.interpolated_traj.points.begin() + nearest_idx, nearest_point);
  control_data.nearest_idx = nearest_idx;
  control_data.target_idx = control_data.nearest_idx; // target_idx will be calculated, we set nearest_idx for now.
  ```
  
  Yes, the variable is not used anywhere, I simplified it. 
  
> `nearest_idx` is not always equivalent to `segment_idx + 1`.

I am not sure I can understand what you mean. I defined it like that because it will be inserted into interpolated_traj (If I understand correctly). I simplified this snipped. Please let me know if there is any problem.

>     * Fix test code. This seems to fail due to the pos/neg definition mismatch.

The test file is updated.

>     * Make a PR to the autoware_launch for new parameters.

I created PR for the parameter update: https://github.com/autowarefoundation/autoware_launch/pull/585
 
>     * Make sure unrelated parameters are not changed in this PR: parameters attached in the screenshot should not change.

Sorry for the changes, I updated.